httpclient/src/test/java/org/apache/http/conn/ssl/TestDefaultHostnameVerifier.java

```diff
@@ -417,4 +419,28 @@ public void testExtractCN() throws Exception {
         }
     }
 
+    @Test
+    public void testMatchDNSName() throws Exception {
+        DefaultHostnameVerifier.matchDNSName(
+                "host.domain.com",
+                Collections.singletonList(SubjectName.DNS("*.domain.com")),
+                publicSuffixMatcher);
+        DefaultHostnameVerifier.matchDNSName(
+                "host.xx",
+                Collections.singletonList(SubjectName.DNS("*.xx")),
+                publicSuffixMatcher);
+        DefaultHostnameVerifier.matchDNSName(
+                "host.appspot.com",
+                Collections.singletonList(SubjectName.DNS("*.appspot.com")),
+                publicSuffixMatcher);
+        DefaultHostnameVerifier.matchDNSName(
+                "demo-s3-bucket.s3.eu-central-1.amazonaws.com",
+                Collections.singletonList(SubjectName.DNS("*.s3.eu-central-1.amazonaws.com")),
+                publicSuffixMatcher);
+        DefaultHostnameVerifier.matchDNSName(
+                "hostname-workspace-1.ops.domain.local",
+                Collections.singletonList(SubjectName.DNS("hostname-workspace-1.ops.domain.local")),
```

Comment:
Call me stupid, but [this](https://publicsuffix.org/list/public_suffix_list.dat) list does not include the `.local` domain and according to [RFC 6762, section 3](https://tools.ietf.org/html/rfc6762#section-3) this hostname is invalid. only a single DNS label is allowed in the `.local` domain.

Reply:
@michael-o I am not sure I understand your objection. The domain name in question might well be invalid but the matching logic should still work, should not it?

Reply:
I agree that the matching logic should work, but I don't think we should use something forbidden in our tests. `hostname-workspace-1.ops.local` should do, shouldn't it for the tests?

Reply:
@michael-o Sure. That I can do.